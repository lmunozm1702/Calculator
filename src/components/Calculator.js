import React from 'react';

class Calculator extends React.PureComponent {
  render() {
    return (
      <div className="App">
        <div className="screen">{0}</div>
        <div className="buttons-container">
          <div className="number"><button type="button" className="button">AC</button></div>
          <div className="number"><button type="button" className="button">+/-</button></div>
          <div className="number"><button type="button" className="button">%</button></div>
          <div className="number operator"><button type="button" className="button">/</button></div>
          <div className="number"><button type="button" className="button">7</button></div>
          <div className="number"><button type="button" className="button">8</button></div>
          <div className="number"><button type="button" className="button">9</button></div>
          <div className="number operator"><button type="button" className="button">X</button></div>
          <div className="number"><button type="button" className="button">4</button></div>
          <div className="number"><button type="button" className="button">5</button></div>
          <div className="number"><button type="button" className="button">6</button></div>
          <div className="number operator"><button type="button" className="button">-</button></div>
          <div className="number"><button type="button" className="button">1</button></div>
          <div className="number"><button type="button" className="button">2</button></div>
          <div className="number"><button type="button" className="button">3</button></div>
          <div className="number operator"><button type="button" className="button">+</button></div>
          <div className="number zero"><button type="button" className="button">0</button></div>
          <div className="number"><button type="button" className="button">.</button></div>
          <div className="number operator"><button type="button" className="button">=</button></div>
        </div>
      </div>
    );
  }
}

export default Calculator;
