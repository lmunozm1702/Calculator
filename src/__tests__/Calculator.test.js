import React from 'react';
import renderer from 'react-test-renderer';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../components/Calculator';

it('renders correctly', () => {
  const calculator = renderer
    .create(<Calculator />)
    .toJSON();
  expect(calculator).toMatchSnapshot();
});

test('user interaction with calc', async () => {
  render(<Calculator url="/calculator" />);
  fireEvent.click(screen.getByText('8'));
  fireEvent.click(screen.getByText('7'));
  expect(screen.getByText('87')).toBeInTheDocument();
});
