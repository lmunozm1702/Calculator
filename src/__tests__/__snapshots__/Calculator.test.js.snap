// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <h3>
    Let's do some math!
  </h3>
  <div
    className="screen"
  >
    0
  </div>
  <div
    className="buttons-container"
  >
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div
      className="number operator"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="number operator"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="number operator"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="number operator"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="number zero"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </div>
    <div
      className="number"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="number operator"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
